addons/mail/static/src/messaging/entity/entity/entity_manager.js

```
@@ -0,0 +1,276 @@
+     * progress.
+     *
+     * @param {mail.messaging.entity.Entity}
+     * @param {mail.messaging.entity.Field} field
+     */
+    registerToComputeField(entity, field) {
+        this._toComputeFields.set(`${entity.localId}--${field.fieldName}`, true);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Make default values of its fields for newly created entity.
+     *
+     * @private
+     * @param {mail.messaging.entity.Entity} entity
+     */
+    _makeDefaults(entity) {
+        for (const field of Object.values(entity.constructor.fields)) {
+            if (field.fieldType === 'attribute') {
+                field.write(entity, field.default, { registerDependents: false });
+            }
+            if (field.fieldType === 'relation') {
+                if (['one2many', 'many2many'].includes(field.relationType)) {
+                    // Ensure X2many relations are arrays by defaults.
+                    field.write(entity, [], { registerDependents: false });
+                } else {
+                    field.write(entity, undefined, { registerDependents: false });
+                }
+            }
+        }
+    }
+
+    /**
+     * Wrap entity that has just been created in a proxy. Proxy is useful for
+     * auto-getting entities when accessing relational fields.
+     *
+     * @private
+     * @param {mail.messaging.entity.Entity} entity
+     * @return {Proxy} proxified entity
+     */
+    _makeProxifiedEntity(entity) {
+        const proxifiedEntity = new Proxy(entity, {
+            get: (target, k) => {
+                if (k === 'constructor') {
+                    return target[k];
+                }
+                const field = target.constructor.fields[k];
+                if (!field) {
+                    return target[k];
```

If `k` is not on target I think I would prefer a throw instead of returning undefined.

---

again, implementation detail on patch on entity make use of `_super` and it dummy store previous `_super` even if it's `undefined`

---

Ok, maybe add what you just said as a comment.